### PR TITLE
Update `@property` attributes to `@cached_property` part 2

### PR DESCRIPTION
**Pull request by UnknownPlatypus**
Followup to #1761

Replace a bunch of `@property` that are now `@cached_property`

This removes 129 entries from the todolist

### CODE REVIEW

**Comment by intgr** [comment]
This is unblocked, the prerequisite PR #1771 is now merged

**Comment by UnknownPlatypus** [comment]
Thanks @intgr , I will rebase my branches later today

File: scripts/stubtest/allowlist.txt [hunk 1]
```diff
@@ -186,3 +187,132 @@ django.urls.resolvers.URLPattern.lookup_str
 django.urls.resolvers.URLResolver.url_patterns
 django.urls.resolvers.URLResolver.urlconf_module
 django.utils.connection.BaseConnectionHandler.settings
+django.contrib.admin.helpers.InlineAdminFormSet.non_form_errors
```
Comment (UnknownPlatypus) [inline scripts/stubtest/allowlist.txt:190]: I've added them here without reordering to avoid a merge conflict with https://github.com/typeddjango/django-stubs/pull/1761

**Comment by UnknownPlatypus** [comment]
@intgr I've rebased after merge of #1761 so that allowlist entries are ordered correctly. Cheers!

File: django-stubs/db/models/expressions.pyi [hunk 2]
```diff
@@ -265,8 +266,8 @@ class OrderBy(Expression):
 
 class Window(SQLiteNumericMixin, Expression):
     template: str
-    contains_aggregate: bool
-    contains_over_clause: bool
+    contains_aggregate: Literal[False]
+    contains_over_clause: Literal[True]
```
Comment (intgr) [inline django-stubs/db/models/expressions.pyi]: Do we care about hinting these as Literal? In most cases `bool` is fine.

If you want to keep this, maybe split to a separate PR.

File: django-stubs/contrib/admin/helpers.pyi [hunk 3]
```diff
@@ -146,7 +147,7 @@ class InlineAdminFormSet:
     def inline_formset_data(self) -> str: ...
     @property
     def forms(self) -> list[BaseForm]: ...
-    @property
+    @cached_property
     def non_form_errors(self) -> Callable[[], ErrorList]: ...
```
Comment (intgr) [inline django-stubs/contrib/admin/helpers.pyi]: This looks like a regular method in Django 4.2. Not sure why it's typehinted so weird. Am I missing something?

File: django-stubs/contrib/gis/db/backends/mysql/features.pyi [hunk 4]
```diff
@@ -14,7 +15,7 @@ class DatabaseFeatures(BaseSpatialFeatures, MySQLDatabaseFeatures):
     supports_num_points_poly: bool
     @property
     def empty_intersection_returns_none(self) -> bool: ...
-    @property
+    @cached_property
     def supports_geometry_field_unique_index(self) -> bool: ...  # type: ignore[override]
-    @property
+    @cached_property
     def django_test_skips(self) -> dict[str, Any]: ...  # type: ignore[override]
```
Comment (intgr) [inline django-stubs/contrib/gis/db/backends/mysql/features.pyi:21]: Doesn't seem to exist in the GIS DatabaseFeatures class, it's in the ordinary MySQL DatabaseFeatures class.

Looks like something that can be done in a separate PR if you're up for it.

**Review by intgr — COMMENTED** [review]
Thanks. A few comments.

File: django-stubs/db/models/expressions.pyi (hunk 2 above)
Comment (UnknownPlatypus) [reply django-stubs/db/models/expressions.pyi]: It probably doesn't add much. I'll revert this

File: django-stubs/contrib/admin/helpers.pyi (hunk 3 above)
Comment (UnknownPlatypus) [reply django-stubs/contrib/admin/helpers.pyi]: Ye that is weird, it's a regular method your right. I'll fix that

File: django-stubs/contrib/gis/db/backends/mysql/features.pyi (hunk 4 above)
Comment (UnknownPlatypus) [reply django-stubs/contrib/gis/db/backends/mysql/features.pyi:21]: They existed before 4.2 (See [4.1 for ex](https://github.com/django/django/blob/stable/4.1.x/django/contrib/gis/db/backends/mysql/features.py) ).

I think it fine to remove them because they are `Classvar` in base classes:
- `django_test_skips` -> https://github.com/django/django/blob/stable/4.2.x/django/db/backends/base/features.py#L366C20-L366C20
- `empty_intersection_returns_none` -> https://github.com/django/django/blob/stable/4.2.x/django/contrib/gis/db/backends/base/features.py#L59

I'll update base types in a separate PR

**Review by intgr — APPROVED** [review]
Thanks, looks good to me
